include/pybind11/common.h

```diff
@@ -509,6 +509,31 @@ template <typename T>
 struct is_input_iterator<T, void_t<decltype(*std::declval<T &>()), decltype(++std::declval<T &>())>>
     : std::true_type {};
 
+template <typename T> using is_function_pointer = bool_constant<
+    std::is_pointer<T>::value && std::is_function<typename std::remove_pointer<T>::type>::value>;
+
+// Extracts the function signature from a function, function pointer or lambda.
+template <typename F, typename SFINAE = void> struct function_signature {};
```

Comment:
This fails to compile because of the `&F::operator()` when `F` is a plain function pointer; any idea for a workaround?